daml-lf/spec/daml-lf-1.rst

```diff
@@ -2431,605 +2380,206 @@ exact output.
-
-      e  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpUpLookupByKeyErr
-      'lookup_by_key' @Mod:T e  ⇓  Err E
-
-      e  ⇓  Ok v
-    —————————————————————————————————————————————————————————————————————— EvExpUpLookupByKey
-      'lookup_by_key' @Mod:T e
-       ⇓
-      Ok ('lookup_by_key' @Mod:T v)
-
-    —————————————————————————————————————————————————————————————————————— EvExpUpTryCatch
-      'try' @τ e₁ 'catch' x. e₂
-       ⇓
-      Ok ('try' @τ e₁ 'catch' x. e₂)
-
-      e  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioPureErr
-      'spure' @τ e  ⇓  Err E
-
-      e  ⇓  Ok v
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioPure
-      'spure' @τ e  ⇓  Ok ('spure' @τ v)
-
-      e₁  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioBindErr
-      'sbind' x₁ : τ₁ ← e₁ 'in' e₂  ⇓  Err E
-
-      e₁  ⇓  Ok v₁
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioBind
-      'sbind' x₁ : τ₁ ← e₁ 'in' e₂
-        ⇓
-      Ok ('sbind' x₁ : τ₁ ← v₁ 'in' e₂)
-
-      e  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioCommitErr1
-      'commit' @τ e u  ⇓  Err E
-
-      e  ⇓  Ok v₁
-      u  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioCommitErr2
-      'commit' @τ e u  ⇓  Err E
-
-      e  ⇓  Ok v₁
-      u  ⇓  Ok v₂
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioCommit
-      'commit' @τ e u  ⇓  Ok ('commit' @τ v₁ v₂)
-
-      e  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioMustFailAtErr1
-      'must_fail_at' @τ e u  ⇓  Err E
-
-      e  ⇓  Ok v₁
-      u  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioMustFailAtErr2
-      'must_fail_at' @τ e u  ⇓  Err E
-
-      e  ⇓  Ok v₁
-      u  ⇓  Ok v₂
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioMustFailAt
-      'must_fail_at' @τ e u  ⇓  Ok ('must_fail_at' @τ v₁ v₂)
-
-      e  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioPassErr
-      'pass' e  ⇓  Err E
-
-      e  ⇓  Ok v
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioPass
-      'pass' e  ⇓  Ok ('pass' v)
-
-      e  ⇓  Err E
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioGetPartyErr
-      'sget_party' e  ⇓  Err E
-
-      e  ⇓  Ok v
-    —————————————————————————————————————————————————————————————————————— EvExpScenarioGetParty
-      'sget_party' e  ⇓  Ok ('sget_party' v)
+      E[ 'interface_observer' @Mod:I ('to_interface' @Mod:I @Mod':T v) ]
+        →
+      Ok E[ e' [x ↦ v] ]
```

Comment:
Could we remove all the context form the rule  above and add two rules 

```suggestion
      e  →  Ok v
   ———————————————————————— 
      E[ e ]  →  Ok E[v] 
```

```suggestion
      e  → Err err
   ———————————————————————— 
      E[ e ]  →  Err e
```

Reply:
Yes, good idea, I think this would be cleaner. I will separate this into two relations -- one that operates only on redexes, and one that operates on  expressions -- to avoid having multiple derivations for the same expression.